node-tests/rollup-funnel-test.js

```diff
@@ -0,0 +1,93 @@
+var fs = require('fs-extra');
+var assert = require('assert');
+var fixturify = require('fixturify');
+var broccoli = require('broccoli-builder');
+var walkSync = require('walk-sync');
+var RollupFunnel = require('../lib/rollup-funnel');
+
+describe('RollupFunnel', function() {
+  var fixtureDir = 'tmp/fixture';
+  beforeEach(function() {
+    fs.mkdirpSync(fixtureDir);
+    fixturify.writeSync(fixtureDir, {
+      'routes.js': 'import buildRoutes from "ember-engines/routes";import foo from "utils/foo";',
+      'utils': {
+        'foo.js': 'import derp from "./derp";',
+        'derp.js': '',
+        'herp.js': ''
+      },
+      'engine.js': 'import herp from "./utils/herp"'
+    });
+  });
+
+  afterEach(function() {
+    fs.removeSync(fixtureDir);
+  });
+
+  describe('build', function() {
+    it('returns a tree of the dependency graph when using include', function() {
+      var rollupFunnel = new RollupFunnel(fixtureDir, {
+        include: true,
+        rollup: {
+          entry: 'routes.js',
+          external: [ 'ember-engines/routes' ]
+        }
+      });
+
+      var builder = new broccoli.Builder(rollupFunnel);
+
+      return builder.build().then(function(result) {
+        var output = walkSync(result.directory);
+        assert.deepEqual(output, [ 'routes.js', 'utils/', 'utils/derp.js', 'utils/foo.js' ]);
+        return builder.cleanup();
+      });
+    });
+
+    it('returns a tree excluding the dependency graph when using exclude', function() {
+      var rollupFunnel = new RollupFunnel(fixtureDir, {
+        exclude: true,
+        rollup: {
+          entry: 'routes.js',
+          external: [ 'ember-engines/routes' ]
+        }
+      });
+
+      var builder = new broccoli.Builder(rollupFunnel);
+
+      return builder.build().then(function(result) {
+        var output = walkSync(result.directory);
+        assert.deepEqual(output, [ 'engine.js', 'utils/', 'utils/herp.js' ]);
+        return builder.cleanup();
```

Comment:
should be called in `afterEach`

likely in the category of "an exploration of an abstraction to hide these details"